extensions/circuit/src/hintstore/mod.rs

```diff
@@ -347,22 +348,31 @@ impl<F: PrimeField32> InstructionExecutor<F> for HintStoreChip<F> {
             opcode,
             a: rd,
             b: num_words_reg,
-            d,
+            c: mem_ptr_reg,
+            d: mem_imm,
             ..
         } = instruction;
-        debug_assert_eq!(d.as_canonical_u32(), RV32_REGISTER_AS);
+
         let local_opcode = HintStoreOpcode::from_usize(opcode.local_opcode_idx(self.air.offset));
 
         let mut wom = self.wom.lock().unwrap();
 
-        memory.increment_timestamp();
-
-        let (num_words, num_words_read) = if local_opcode == HINT_STOREW {
-            (1, None)
+        let (rd, num_words, num_words_read, mem_ptr) = if local_opcode == HINT_STOREW {
+            memory.increment_timestamp();
```

Comment:
i think its clearer if this moves to the `else` of the "write" if later in the fn (the else here does not do any memory/timestamp operation)

Reply:
removed this since we don't have ram operations in this case anyway